### PR TITLE
Stdin support

### PR DESCRIPTION
**Pull request by nikhilm**
@vlopatkin Please review. Ensures compatibility with some upcoming changes to how payload is delivered to images.

### CODE REVIEW

File: images/java/src/main/java/io/iron/lambda/Launcher.java [hunk 1]
```diff
@@ -15,13 +15,29 @@
     public static void main(String[] args) {
         String handler = args[0];
         String payload = "";
-        try {
-            String file = System.getenv("PAYLOAD_FILE");
-            if (file != null) {
+        String file = System.getenv("PAYLOAD_FILE");
+        if (file != null) {
```
Comment (ccirello) [inline images/java/src/main/java/io/iron/lambda/Launcher.java]: What happens if `PAYLOAD_FILE` is set to empty? I mean, the env var is set, but the content is empty.

**Comment by nikhilm** [comment]
Exception and fail. Should we treat that as unset?

**Comment by ccirello** [comment]
> Should we treat that as unset?

I think so.

**Comment by nikhilm** [comment]
Done

**Comment by ccirello** [comment]
LGTM - now we wait for @vlopatkin review.

File: images/python/bootstrap.py [hunk 2]
```diff
@@ -234,12 +229,17 @@ def filter(self, record):
     stopWithError("Function name is not defined")
 
 try:
-    with file(payloadFileName) as f:
+    if payloadFileName:
+        stopWithError("No payload present")
```
Comment (vlopatkin) [inline images/python/bootstrap.py]: `if not os.path.isfile(payloadFileName):
    stopWithError("No payload present")`
?

File: images/python/bootstrap.py [hunk 3]
```diff
@@ -234,12 +229,17 @@ def filter(self, record):
...
+        payloadFile = sys.stdin
+
+    with payloadFile as f:
         payload = f.read()
-except:
-    stopWithError("Failed to read {payloadFileName}".format(payloadFileName=payloadFileName))
 
-debugging and print ('payload loaded')
+except:
+    stopWithError("Failed to read {payloadFile}".format(payloadFileName=(payloadFileName and payloadFileName or '<stdin>')))
```
Comment (vlopatkin) [inline images/python/bootstrap.py]: "Failed to read {payloadFileName}"?

Comment (vlopatkin) [inline images/python/bootstrap.py]: `(payloadFileName or '<stdin>')`?

File: images/python/bootstrap.py (hunk 2 above)
Comment (nikhilm) [inline images/python/bootstrap.py]: file not existing will be captured by the exception. Modifying script to do that. Thanks for catching these errors. Tested all error combinations now (env undefined, env empty, env not a real file, env is bad json, stdin is bad json)

File: images/python/bootstrap.py (hunk 3 above)
Comment (vlopatkin) [inline images/python/bootstrap.py]: but is does not need to be `x and x or y`
https://trinket.io/python/72f98f886a

**Comment by vlopatkin** [comment]
LGTM

**Comment by vlopatkin** [comment]
"Only those with write access to this repository can merge pull requests."

**Pull request merged**